### PR TITLE
[feat-5588]: refactor containers in detail panel

### PR DESCRIPTION
**Pull request by vdegraaf**
Ticket: [SIG-5588](https://gemeente-amsterdam.atlassian.net/browse/SIG-5580)

### Changes
- Created a mapper (which can be used for lights and containers) in WfsLayer that maps features to a selectableFeature model
- Use this model in the AssetList
- Removed the specific logic from useSelectionProps
- Edit and update test cases

[SIG-5580]: https://gemeente-amsterdam.atlassian.net/browse/SIG-5580?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[SIG-5588]: https://gemeente-amsterdam.atlassian.net/browse/SIG-5588?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/signals/incident/components/form/MapSelectors/Asset/AssetList/AssetList.tsx [hunk 1]
```diff
@@ -65,10 +62,10 @@ const AssetList: FunctionComponent<AssetListProps> = ({
             ) && (
               <ListDescription>
                 {`Er zijn geen ${
```
Comment (OscarBakker) [inline src/signals/incident/components/form/MapSelectors/Asset/AssetList/AssetList.tsx:64]: ja better!

File: src/signals/incident/components/form/MapSelectors/Asset/AssetList/hooks/useSelectionProps.tsx [hunk 2]
```diff
@@ -33,50 +32,25 @@ export const useSelectionProps = ({
...
   const featureStatusType = featureStatusTypes.find(
     ({ typeValue }) => typeValue === status
   )
-
-  const label = isTemplateString(description)
-    ? parseTemplateString(description, feature.properties)
-    : [description, id].filter(Boolean).join(' - ')
-
   const item: Item = {
-    id: `${coordinates.lat}.${coordinates.lng}.${feature.properties.created_at}`,
```
Comment (OscarBakker) [inline src/signals/incident/components/form/MapSelectors/Asset/AssetList/hooks/useSelectionProps.tsx:54]: cleanup!

File: src/signals/incident/components/form/MapSelectors/Asset/types.ts [hunk 3]
```diff
@@ -20,12 +18,12 @@ export interface AssetSelectValue {
   message?: string
   meta: Meta
   removeItem: (item?: Item) => void
-  selectableFeatures?: FeatureCollection
+  selectableFeatures?: SelectableFeature[]
```
Comment (OscarBakker) [inline src/signals/incident/components/form/MapSelectors/Asset/types.ts:21]: maybe move to bottom

File: src/signals/incident/components/form/MapSelectors/Asset/Selector_v2_removeafterfinishepic5440/WfsLayer/utils/test/mock-feature-types.ts [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        iconSize: [40, 40],
+      },
+      iconUrl: '/assets/images/groen_water/oak.svg',
+    },
+    idField: 'OBJECTID',
+    typeValue: 'Eikenboom',
+    typeField: '',
+  },
+  {
+    // idField is not really returned
```
Comment (OscarBakker) [inline src/signals/incident/components/form/MapSelectors/Asset/Selector_v2_removeafterfinishepic5440/WfsLayer/utils/test/mock-feature-types.ts:223]: but what is it?

File: src/signals/incident/components/form/MapSelectors/Asset/Selector_v2_removeafterfinishepic5440/WfsLayer/utils/map-data-to-selectable-feature.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        const coordinates = featureToCoordinates(feature?.geometry as Geometrie)
+
+        return {
+          coordinates,
+          description: description,
+          id: id_number,
+          label: label,
+          type: typeValue,
+        }
+      })
```
Comment (OscarBakker) [inline src/signals/incident/components/form/MapSelectors/Asset/Selector_v2_removeafterfinishepic5440/WfsLayer/utils/map-data-to-selectable-feature.ts:52]: could be refactored as function mapFeatureToSelectable and then placed under  mapDataToSelectableFeature

```
switch (objectType) {
    case FeatureTypes.CONTAINER:
    case FeatureTypes.PUBLIC_LIGHTS:
      return features.map(mapFeatureToSelectable
```

just a suggestion

**Review by OscarBakker — APPROVED** [review]
some suggestions left

File: src/signals/incident/components/form/MapSelectors/Asset/Selector_v2_removeafterfinishepic5440/WfsLayer/utils/map-data-to-selectable-feature.ts (hunk 5 above)
Comment (vdegraaf) [reply src/signals/incident/components/form/MapSelectors/Asset/Selector_v2_removeafterfinishepic5440/WfsLayer/utils/map-data-to-selectable-feature.ts:52]: I might do that later, when caterpillars is also implemented and see what the overlap is.

**Pull request merged**